File: QTD2.Domain/Entities/Core/SegmentObjective_Link.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using QTD2.Domain.Entities.Common;

namespace QTD2.Domain.Entities.Core
{
    public class SegmentObjective_Link : Entity
    {
        public int SegmentId { get; set; }

        public int Order { get; set; }

        public int? TaskId { get; set; }

        public int? EnablingObjectiveId { get; set; }

        public int? CustomEOId { get; set; }

        public virtual Segment Segment { get; set; }

        public virtual Task Task { get; set; }

        public virtual EnablingObjective EnablingObjective { get; set; }

        public virtual CustomEnablingObjective CustomEnablingObjective { get; set; }

        public SegmentObjective_Link(Segment segment, Task task, EnablingObjective enablingObjective, CustomEnablingObjective co)
        {
            Segment = segment;
            Task = task;
            EnablingObjective = enablingObjective;
            SegmentId = segment.Id;
            TaskId = task == null ? null:task.Id;
            EnablingObjectiveId = enablingObjective?.Id;
            CustomEOId = co?.Id;
        }

        public SegmentObjective_Link()
        {
        }

        public void SetOrder(int order)
        {
            Order = order;
        }
    }
}

File: QTD.SPA/src/app/_DtoModels/Segment_ObjectiveLink/Segment_ObjectiveLink.ts
import { CustomEnablingObjective } from "../CustomEnablingObjective/CustomEnablingObjective";
import { EnablingObjective } from "../EnablingObjective/EnablingObjective";
import { Entity } from "../Entity";
import { Task } from "../Task/Task";

export class Segment_ObjectiveLink extends Entity {
  segmentId!: any;
  taskIds!: any;
  enablingObjectiveIds!: any;
  task!:Task;
  enablingObjective!:EnablingObjective;
  customEnablingObjective!:CustomEnablingObjective;
  order?:any;
}
